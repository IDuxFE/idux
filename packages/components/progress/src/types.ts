/**
 * @license
 *
 * Use of this source code is governed by an MIT-style license that can be
 * found in the LICENSE file at https://github.com/IDuxFE/idux/blob/main/LICENSE
 */

import type { IxInnerPropTypes, IxPublicPropTypes } from '@idux/cdk/utils'
import type { DefineComponent, HTMLAttributes, VNode, VNodeChild } from 'vue'

import { IxPropTypes } from '@idux/cdk/utils'

export type ProgressSize = 'sm' | 'md' | 'lg'
export type ProgressFormat = (percent: number, successPercent?: number) => VNodeChild
export type ProgressType = 'line' | 'circle' | 'dashboard'
export type ProgressGapPositionType = 'top' | 'bottom' | 'left' | 'right'
export type ProgressStatus = 'normal' | 'success' | 'exception' | 'active'
export type ProgressStrokeLinecap = 'round' | 'square'
export type StringGradients = { [percentage: string]: string }
export type FromToGradients = { from: string; to: string }
export type ProgressGradient = { direction?: string } & (StringGradients | FromToGradients)

export interface ProgressSuccess {
  percent?: number | string
  strokeColor?: string
}

export interface ConvertProgressSuccess extends ProgressSuccess {
  percent: number
}

export interface ProgressIcons {
  success: string | VNode
  exception: string | VNode
}

export const progressStatus = ['normal', 'success', 'exception', 'active'] as const

export const progressProps = {
  type: IxPropTypes.oneOf<ProgressType>(['line', 'circle', 'dashboard']).def('line'),
  format: IxPropTypes.func<ProgressFormat>(),
  percent: IxPropTypes.oneOfType([String, Number]).def(0),
  status: IxPropTypes.oneOf<ProgressStatus>(['normal', 'success', 'exception', 'active']),
  hideInfo: IxPropTypes.bool.def(false),
  success: IxPropTypes.object<ProgressSuccess>(),
  trailColor: IxPropTypes.string,
  strokeColor: IxPropTypes.oneOfType([String, IxPropTypes.object<ProgressGradient>()]),
  strokeLinecap: IxPropTypes.oneOf<ProgressStrokeLinecap>(['round', 'square']),
  strokeWidth: IxPropTypes.oneOfType([String, Number]),
  gapDegree: IxPropTypes.oneOfType([String, Number]),
  gapPosition: IxPropTypes.oneOf<ProgressGapPositionType>(['top', 'bottom', 'left', 'right']),
  width: IxPropTypes.oneOfType([String, Number]).def(132),
  size: IxPropTypes.oneOf<ProgressSize>(['sm', 'md', 'lg']),
  icons: IxPropTypes.object<ProgressIcons>(),
}

export type ProgressProps = IxInnerPropTypes<typeof progressProps>
export type ProgressPublicProps = IxPublicPropTypes<typeof progressProps>
export type ProgressComponent = DefineComponent<Omit<HTMLAttributes, keyof ProgressPublicProps> & ProgressPublicProps>
export type ProgressInstance = InstanceType<DefineComponent<ProgressProps>>
