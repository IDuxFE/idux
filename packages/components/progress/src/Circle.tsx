/**
 * @license
 *
 * Use of this source code is governed by an MIT-style license that can be
 * found in the LICENSE file at https://github.com/IDuxFE/idux/blob/main/LICENSE
 */

import { ComputedRef, computed, defineComponent, inject } from 'vue'

import { isObject } from 'lodash-es'

import { convertNumber, uniqueId } from '@idux/cdk/utils'

import ProgressInfo from './ProgressInfo'
import { progressContext } from './tokens'
import { ProgressGapPositionType, ProgressProps, StringGradients } from './types'
import { handleCircleGradient } from './util'

export interface CalcSharedProperties {
  isGradient: boolean
  percent: number
  linearGradientId: string
  radius: number
  gapPosition: ProgressGapPositionType
  len: number
  gapDegree: number
}

const defaultStrokeWidth = 6

export default defineComponent({
  name: 'IxProgressCircle',
  setup(_, { slots }) {
    const { props, mergedPrefixCls, mergedStrokeLinecap, percent, successPercent } = inject(progressContext)!

    const circleMergedPrefixCls = computed(() => `${mergedPrefixCls.value}-circle`)

    const strokeWidth = computed(() => convertNumber(props.strokeWidth, defaultStrokeWidth))
    const isGradient = computed(() => isObject(props.strokeColor))
    const linearGradientId = computed(() => `${mergedPrefixCls.value}-gradient-${uniqueId()}`)

    const calcSharedProperties = computed<CalcSharedProperties>(() => {
      const isCircle = props.type === 'circle'
      const radius = 50 - strokeWidth.value / 2
      return {
        isGradient: isGradient.value,
        percent: percent.value,
        linearGradientId: linearGradientId.value,
        radius,
        gapPosition: props.gapPosition ?? (isCircle ? 'top' : 'bottom'),
        len: Math.PI * 2 * radius,
        gapDegree: convertNumber(props.gapDegree ?? (isCircle ? 0 : 75)),
      }
    })
    const circleGradient = computed(() => {
      return isGradient.value ? handleCircleGradient(props.strokeColor as StringGradients) : []
    })
    const pathString = usePathString(calcSharedProperties)
    const trailPathStyle = useTrailPathStyle(calcSharedProperties)
    const strokePath = useCirclePath(calcSharedProperties, circleMergedPrefixCls, props, percent, successPercent)

    const trailPathAttr = computed(() => ({
      stroke: props.trailColor,
      'fill-opacity': '0',
      'stroke-linecap': mergedStrokeLinecap.value,
      'stroke-width': strokeWidth.value,
      d: pathString.value,
    }))
    const strokePathAttr = computed(() => ({
      'fill-opacity': '0',
      'stroke-linecap': mergedStrokeLinecap.value,
      'stroke-width': percent.value ? strokeWidth.value : 0,
      d: pathString.value,
    }))

    const circleClasses = computed(() => {
      const prefixCls = circleMergedPrefixCls.value

      return {
        [prefixCls]: true,
        [`${prefixCls}-gradient`]: isGradient.value,
      }
    })
    const circleStyle = computed(() => ({
      width: props.width && `${props.width}px`,
      height: props.width && `${props.width}px`,
      fontSize: props.width && `${convertNumber(props.width) * 0.15 + 6}px`,
    }))

    const renderDefs = () => {
      if (!isGradient.value) {
        return null
      }

      return (
        <defs>
          <linearGradient id={linearGradientId.value} x1="100%" y1="0%" x2="0%" y2="0%">
            {circleGradient.value.map(gradient => (
              <stop offset={gradient.offset} stop-color={gradient.color}></stop>
            ))}
          </linearGradient>
        </defs>
      )
    }

    const renderStrokePath = () =>
      strokePath.value.map(p => (
        <path
          class={[`${circleMergedPrefixCls.value}-path`, ...p.strokeClasses]}
          style={p.strokePathStyle}
          stroke={p.stroke}
          {...strokePathAttr.value}
        ></path>
      ))

    return () => (
      <div class={circleClasses.value} style={circleStyle.value}>
        <svg viewBox="0 0 100 100">
          {renderDefs()}
          <path
            class={`${circleMergedPrefixCls.value}-trail`}
            style={trailPathStyle.value}
            {...trailPathAttr.value}
          ></path>
          {renderStrokePath()}
        </svg>
        <ProgressInfo v-slots={slots} />
      </div>
    )
  },
})

function usePathString(calcSharedProperties: ComputedRef<CalcSharedProperties>): ComputedRef<string> {
  return computed(() => {
    const { radius, gapPosition } = calcSharedProperties.value
    let beginPositionX = 0
    let beginPositionY = -radius
    let endPositionX = 0
    let endPositionY = radius * -2

    switch (gapPosition) {
      case 'left':
        beginPositionX = -radius
        beginPositionY = 0
        endPositionX = radius * 2
        endPositionY = 0
        break
      case 'right':
        beginPositionX = radius
        beginPositionY = 0
        endPositionX = radius * -2
        endPositionY = 0
        break
      case 'bottom':
        beginPositionY = radius
        endPositionY = radius * 2
        break
      default:
    }

    return `M 50,50 m ${beginPositionX},${beginPositionY}
       a ${radius},${radius} 0 1 1 ${endPositionX},${-endPositionY}
       a ${radius},${radius} 0 1 1 ${-endPositionX},${endPositionY}`
  })
}

function useTrailPathStyle(calcSharedProperties: ComputedRef<CalcSharedProperties>): ComputedRef<{
  strokeDasharray: string
  strokeDashoffset: string
  transition: string
}> {
  return computed(() => {
    const { len, gapDegree } = calcSharedProperties.value
    return {
      strokeDasharray: `${len - gapDegree}px ${len}px`,
      strokeDashoffset: `-${gapDegree / 2}px`,
      transition: 'stroke-dashoffset .3s ease 0s, stroke-dasharray .3s ease 0s, stroke .3s',
    }
  })
}

function useCirclePath(
  calcSharedProperties: ComputedRef<CalcSharedProperties>,
  circleMergedPrefixCls: ComputedRef<string>,
  props: ProgressProps,
  percent: ComputedRef<number>,
  successPercent: ComputedRef<number>,
) {
  return computed(() => {
    const { gapDegree, len, isGradient, linearGradientId } = calcSharedProperties.value
    const strokeProgress = successPercent.value > 0 ? [successPercent.value, percent.value] : [percent.value]
    const successColor = props.success?.strokeColor

    return strokeProgress
      .map((value, index) => {
        const hasSuccessPercent = strokeProgress.length > 1 && index === 0
        return {
          stroke: isGradient && !hasSuccessPercent ? `url(#${linearGradientId})` : undefined,
          strokeClasses: [
            !isGradient && hasSuccessPercent ? `${circleMergedPrefixCls.value}-success` : '',
            isGradient ? '' : `${circleMergedPrefixCls.value}-bg`,
          ],
          strokePathStyle: {
            stroke: !isGradient ? (hasSuccessPercent ? successColor : (props.strokeColor as string)) : undefined,
            transition:
              'stroke-dashoffset .3s ease 0s, stroke-dasharray .3s ease 0s, stroke .3s, stroke-width .06s ease .3s',
            strokeDasharray: `${(value / 100) * (len - gapDegree)}px ${len}px`,
            strokeDashoffset: `-${gapDegree / 2}px`,
          },
        }
      })
      .reverse()
  })
}
