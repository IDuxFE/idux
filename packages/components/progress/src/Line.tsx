/**
 * @license
 *
 * Use of this source code is governed by an MIT-style license that can be
 * found in the LICENSE file at https://github.com/IDuxFE/idux/blob/main/LICENSE
 */

import { computed, defineComponent, inject } from 'vue'

import { isObject } from 'lodash-es'

import ProgressInfo from './ProgressInfo'
import { useProps } from './composables/useProps'
import { progressContext } from './tokens'
import { handleGradient } from './util'

export default defineComponent({
  name: 'IxProgressLine',
  setup() {
    const { props, config, mergedPrefixCls, percent, formattedSuccess } = inject(progressContext)!
    const computedProps = useProps(props, config)
    const lineMergedPrefixCls = computed(() => `${mergedPrefixCls.value}-line`)

    const lineClasses = computed(() => {
      const prefixCls = lineMergedPrefixCls.value

      return {
        [prefixCls]: true,
        [`${prefixCls}-${computedProps.value.size}`]: true,
        [`${prefixCls}-round`]: computedProps.value.strokeLinecap === 'round',
      }
    })
    const successStyle = computed(() => ({
      height: computedProps.value.strokeWidth && `${computedProps.value.strokeWidth}px`,
      width: `${formattedSuccess.value.percent ?? 0}%`,
      background: formattedSuccess.value.strokeColor ?? '',
    }))
    const bgStyle = computed(() => ({
      height: computedProps.value.strokeWidth && `${computedProps.value.strokeWidth}px`,
      width: `${percent.value}%`,
      background: isObject(computedProps.value.strokeColor)
        ? handleGradient(computedProps.value.strokeColor)
        : computedProps.value.strokeColor ?? '',
    }))

    return () => {
      const prefixCls = lineMergedPrefixCls.value

      return (
        <div class={lineClasses.value}>
          <div class={`${prefixCls}-outer`}>
            <div class={`${prefixCls}-inner`}>
              {computedProps.value.success?.percent && (
                <div class={`${prefixCls}-success-bg`} style={successStyle.value}></div>
              )}
              <div class={`${prefixCls}-bg`} style={bgStyle.value}></div>
            </div>
          </div>
          <ProgressInfo />
        </div>
      )
    }
  },
})
