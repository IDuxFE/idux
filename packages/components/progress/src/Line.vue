<template>
  <div class="ix-progress-line" :class="lineClasses">
    <div class="ix-progress-outer">
      <div class="ix-progress-inner">
        <div v-if="success.percent" class="ix-progress-success-bg" :style="successStyle"></div>
        <div class="ix-progress-bg" :style="bgStyle"></div>
      </div>
    </div>
    <div v-if="!hideInfo" class="ix-progress-text">
      <slot>
        <template v-if="showFormat">{{ formattedText }}</template>
        <IxIcon v-else-if="showSuccessIcon" class="ix-progress-success-icon" name="check-circle-filled" />
        <IxIcon v-else-if="showExceptionIcon" class="ix-progress-exception-icon" name="close-circle-filled" />
      </slot>
    </div>
  </div>
</template>
<script lang="ts">
import { computed, defineComponent } from 'vue'

import { isObject } from 'lodash-es'

import { convertNumber } from '@idux/cdk/utils'
import { useGlobalConfig } from '@idux/components/config'
import { IxIcon } from '@idux/components/icon'

import { convertProgressProps } from './types'
import { useInfo, useSmallSize, useStatus, useStatusClasses } from './useCommonLogic'
import { handleGradient } from './util'

const defaultStrokeWidth = 8
const defaultSmallStrokeWidth = 6

export default defineComponent({
  name: 'IxProgressLine',
  components: { IxIcon },
  props: convertProgressProps,
  setup(props) {
    const progressConfig = useGlobalConfig('progress')
    const isSmallSize = useSmallSize(props, progressConfig)
    const strokeWidth = computed(() =>
      convertNumber(props.strokeWidth ?? (isSmallSize.value ? defaultSmallStrokeWidth : defaultStrokeWidth)),
    )
    const progressStatus = useStatus(props)
    const statusClass = useStatusClasses(progressStatus)
    const { formattedText, showSuccessIcon, showExceptionIcon, showFormat } = useInfo(
      props,
      progressConfig,
      progressStatus,
    )

    const lineClasses = computed(() => {
      return [
        statusClass.value,
        props.hideInfo ? '' : 'ix-progress-show-info',
        props.strokeLinecap === 'round' ? 'ix-progress-round' : '',
        isSmallSize.value ? 'ix-progress-sm' : '',
      ]
    })
    const successStyle = computed(() => ({
      height: `${strokeWidth.value}px`,
      width: `${props.success.percent}%`,
      background: props.success?.strokeColor ?? '',
    }))
    const bgStyle = computed(() => ({
      height: `${strokeWidth.value}px`,
      width: `${props.percent}%`,
      background: isObject(props.strokeColor) ? handleGradient(props.strokeColor) : props.strokeColor ?? '',
    }))

    return {
      lineClasses,
      successStyle,
      bgStyle,
      formattedText,
      showFormat,
      showSuccessIcon,
      showExceptionIcon,
    }
  },
})
</script>
