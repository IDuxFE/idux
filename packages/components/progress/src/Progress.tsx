/**
 * @license
 *
 * Use of this source code is governed by an MIT-style license that can be
 * found in the LICENSE file at https://github.com/IDuxFE/idux/blob/main/LICENSE
 */

import { computed, defineComponent, provide } from 'vue'

import { useGlobalConfig } from '@idux/components/config'

import Circle from './Circle'
import Line from './Line'
import { useStatus } from './composables/useStatus'
import { progressContext } from './tokens'
import { progressProps } from './types'
import { convertPercent } from './util'

export default defineComponent({
  name: 'IxProgress',
  props: progressProps,
  setup(props, { slots }) {
    const config = useGlobalConfig('progress')
    const common = useGlobalConfig('common')
    const mergedPrefixCls = computed(() => `${common.prefixCls}-progress`)

    const percent = computed(() => convertPercent(props.percent))
    const formattedSuccess = computed(() => ({ ...props.success, percent: convertPercent(props.success?.percent) }))

    const status = useStatus(props, percent, formattedSuccess)

    const classes = computed(() => {
      const prefixCls = mergedPrefixCls.value
      return {
        [prefixCls]: true,
        [`${prefixCls}-${status.value}`]: !!status.value,
      }
    })

    provide(progressContext, {
      props,
      config,
      slots,
      mergedPrefixCls,
      percent,
      formattedSuccess,
      status,
    })

    return () => {
      const ProgressComponent = props.type === 'line' ? Line : Circle
      return <ProgressComponent class={classes.value} />
    }
  },
})
