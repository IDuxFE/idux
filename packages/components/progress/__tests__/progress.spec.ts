import { mount, MountingOptions, VueWrapper } from '@vue/test-utils'
import { DefineComponent } from 'vue'
import IxProgress from '../src/Progress.vue'
import { ProgressProps } from '../src/types'

describe('Progress.vue', () => {
  let ProgressMount: (
    options?: MountingOptions<Partial<ProgressProps>>,
  ) => VueWrapper<InstanceType<DefineComponent<ProgressProps>>>

  beforeEach(() => {
    ProgressMount = (options = {}) => {
      return mount<ProgressProps>(IxProgress, {
        ...options,
      })
    }
  })

  test('render line progress work', () => {
    const wrapper = ProgressMount({ props: { percent: 50 } })
    expect(wrapper.html()).toMatchSnapshot()
  })

  test('render circle progress work', () => {
    const wrapper = ProgressMount({ props: { type: 'circle', percent: 50 } })
    expect(wrapper.html()).toMatchSnapshot()
  })

  test('render dashboard progress work', () => {
    const wrapper = ProgressMount({ props: { type: 'dashboard', percent: 50 } })
    expect(wrapper.html()).toMatchSnapshot()
  })

  test('render info content', async () => {
    const wrapper = ProgressMount({ props: { percent: 0 } })
    expect(wrapper.text()).toBe('0%')

    await wrapper.setProps({ percent: 50 })
    expect(wrapper.text()).toBe('50%')

    await wrapper.setProps({ percent: 100 })
    expect(wrapper.text()).toBe('')
  })

  test('render out-of-range info content', async () => {
    const wrapper = ProgressMount({ props: { percent: -10 } })
    expect(wrapper.text()).toBe('0%')

    await wrapper.setProps({ percent: 110 })
    expect(wrapper.text()).toBe('')
  })

  test('render info content with hideInfo', () => {
    const wrapper = ProgressMount({ props: { percent: 50, hideInfo: true } })
    expect(wrapper.classes('ix-progress-text')).toBe(false)
  })

  test('render info content by format', () => {
    const wrapper = ProgressMount({ props: { percent: 50, format: percent => `format: ${percent}` } })
    expect(wrapper.text()).toBe('format: 50')
  })

  test('render info content by slot', () => {
    const text = 'render by slot'
    const wrapper = ProgressMount({ props: { percent: 50 }, slots: { format: text } })
    expect(wrapper.text()).toBe(text)
  })

  test('status work', async () => {
    const wrapper = ProgressMount({ props: { percent: 50, status: 'normal' } })
    expect(wrapper.classes('ix-progress-status-normal')).toBe(true)

    await wrapper.setProps({ status: 'success' })
    expect(wrapper.classes('ix-progress-status-success')).toBe(true)

    await wrapper.setProps({ status: 'exception' })
    expect(wrapper.classes('ix-progress-status-exception')).toBe(true)

    await wrapper.setProps({ status: 'active' })
    expect(wrapper.classes('ix-progress-status-active')).toBe(true)
  })

  test('should show success status when percent = 100', () => {
    const wrapper = ProgressMount({ props: { percent: 100 } })
    expect(wrapper.classes('ix-progress-status-success')).toBe(true)
  })

  test('strokeLinecap work', async () => {
    const wrapper = ProgressMount({ props: { percent: 50, strokeLinecap: 'round' } })
    expect(wrapper.classes('ix-progress-round')).toBe(true)

    await wrapper.setProps({ strokeLinecap: 'square' })
    expect(wrapper.classes('ix-progress-round')).toBe(false)
  })

  test('strokeColor work', async () => {
    const wrapperLine = ProgressMount({ props: { percent: 50, strokeColor: 'red' } })
    const wrapperCircle = ProgressMount({
      props: { type: 'circle', percent: 50, strokeColor: { '0%': '#108ee9', '100%': '#87d068' } },
    })
    expect(wrapperLine.html()).toMatchSnapshot()
    expect(wrapperCircle.html()).toMatchSnapshot()

    await wrapperLine.setProps({ strokeColor: { from: '#108ee9', to: '#87d068' } })
    expect(wrapperLine.html()).toMatchSnapshot()

    await wrapperLine.setProps({ strokeColor: { '0%': '#108ee9', '100%': '#87d068' } })
    expect(wrapperLine.html()).toMatchSnapshot()
  })

  test('render successColor progress', () => {
    const wrapper = ProgressMount({ props: { percent: 50, success: { percent: 20, strokeColor: 'red' } } })
    expect(wrapper.html()).toMatchSnapshot()
  })

  test('trailColor work', () => {
    const wrapper = ProgressMount({ props: { percent: 50, trailColor: 'red' } })
    expect(wrapper.html()).toMatchSnapshot()
  })

  test('gapPosition work', async () => {
    const wrapper = ProgressMount({ props: { type: 'dashboard', percent: 50, trailColor: 'red' } })
    expect(wrapper.html()).toMatchSnapshot()

    await wrapper.setProps({ gapPosition: 'left' })
    expect(wrapper.html()).toMatchSnapshot()

    await wrapper.setProps({ gapPosition: 'right' })
    expect(wrapper.html()).toMatchSnapshot()
  })
})
