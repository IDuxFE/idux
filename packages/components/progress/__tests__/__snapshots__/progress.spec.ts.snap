// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Progress.vue gapPosition work 1`] = `
"<div class=\\"ix-progress-circle ix-progress-status-normal ix-progress\\" style=\\"width: 132px; height: 132px; font-size: 25.8px;\\"><svg viewBox=\\"0 0 100 100\\">
    <!--v-if-->
    <path class=\\"ix-progress-circle-trail\\" stroke=\\"#f5f5f5\\" fill-opacity=\\"0\\" stroke-linecap=\\"round\\" stroke-width=\\"6\\" d=\\"M 50,50 m 0,47
       a 47,47 0 1 1 0,-94
       a 47,47 0 1 1 0,94\\" style=\\"stroke-dasharray: 220.30970943744057px 295.3097094374406px; stroke-dashoffset: -37.5px; transition: stroke-dashoffset .3s ease 0s, stroke-dasharray .3s ease 0s, stroke .3s;\\"></path>
    <path class=\\"ix-progress-circle-path ix-progress-circle-bg\\" fill-opacity=\\"0\\" stroke-linecap=\\"round\\" stroke-width=\\"6\\" d=\\"M 50,50 m 0,47
       a 47,47 0 1 1 0,-94
       a 47,47 0 1 1 0,94\\" style=\\"transition: stroke-dashoffset .3s ease 0s, stroke-dasharray .3s ease 0s, stroke .3s, stroke-width .06s ease .3s; stroke-dasharray: 110.15485471872029px 295.3097094374406px; stroke-dashoffset: -37.5px;\\"></path>
  </svg>
  <div class=\\"ix-progress-text\\">50%</div>
</div>"
`;

exports[`Progress.vue gapPosition work 2`] = `
"<div class=\\"ix-progress-circle ix-progress-status-normal ix-progress\\" style=\\"width: 132px; height: 132px; font-size: 25.8px;\\"><svg viewBox=\\"0 0 100 100\\">
    <!--v-if-->
    <path class=\\"ix-progress-circle-trail\\" stroke=\\"#f5f5f5\\" fill-opacity=\\"0\\" stroke-linecap=\\"round\\" stroke-width=\\"6\\" d=\\"M 50,50 m -47,0
       a 47,47 0 1 1 94,0
       a 47,47 0 1 1 -94,0\\" style=\\"stroke-dasharray: 220.30970943744057px 295.3097094374406px; stroke-dashoffset: -37.5px; transition: stroke-dashoffset .3s ease 0s, stroke-dasharray .3s ease 0s, stroke .3s;\\"></path>
    <path class=\\"ix-progress-circle-path ix-progress-circle-bg\\" fill-opacity=\\"0\\" stroke-linecap=\\"round\\" stroke-width=\\"6\\" d=\\"M 50,50 m -47,0
       a 47,47 0 1 1 94,0
       a 47,47 0 1 1 -94,0\\" style=\\"transition: stroke-dashoffset .3s ease 0s, stroke-dasharray .3s ease 0s, stroke .3s, stroke-width .06s ease .3s; stroke-dasharray: 110.15485471872029px 295.3097094374406px; stroke-dashoffset: -37.5px;\\"></path>
  </svg>
  <div class=\\"ix-progress-text\\">50%</div>
</div>"
`;

exports[`Progress.vue gapPosition work 3`] = `
"<div class=\\"ix-progress-circle ix-progress-status-normal ix-progress\\" style=\\"width: 132px; height: 132px; font-size: 25.8px;\\"><svg viewBox=\\"0 0 100 100\\">
    <!--v-if-->
    <path class=\\"ix-progress-circle-trail\\" stroke=\\"#f5f5f5\\" fill-opacity=\\"0\\" stroke-linecap=\\"round\\" stroke-width=\\"6\\" d=\\"M 50,50 m 47,0
       a 47,47 0 1 1 -94,0
       a 47,47 0 1 1 94,0\\" style=\\"stroke-dasharray: 220.30970943744057px 295.3097094374406px; stroke-dashoffset: -37.5px; transition: stroke-dashoffset .3s ease 0s, stroke-dasharray .3s ease 0s, stroke .3s;\\"></path>
    <path class=\\"ix-progress-circle-path ix-progress-circle-bg\\" fill-opacity=\\"0\\" stroke-linecap=\\"round\\" stroke-width=\\"6\\" d=\\"M 50,50 m 47,0
       a 47,47 0 1 1 -94,0
       a 47,47 0 1 1 94,0\\" style=\\"transition: stroke-dashoffset .3s ease 0s, stroke-dasharray .3s ease 0s, stroke .3s, stroke-width .06s ease .3s; stroke-dasharray: 110.15485471872029px 295.3097094374406px; stroke-dashoffset: -37.5px;\\"></path>
  </svg>
  <div class=\\"ix-progress-text\\">50%</div>
</div>"
`;

exports[`Progress.vue render circle progress work 1`] = `
"<div class=\\"ix-progress-circle ix-progress-status-normal ix-progress\\" style=\\"width: 132px; height: 132px; font-size: 25.8px;\\"><svg viewBox=\\"0 0 100 100\\">
    <!--v-if-->
    <path class=\\"ix-progress-circle-trail\\" stroke=\\"#f5f5f5\\" fill-opacity=\\"0\\" stroke-linecap=\\"round\\" stroke-width=\\"6\\" d=\\"M 50,50 m 0,-47
       a 47,47 0 1 1 0,94
       a 47,47 0 1 1 0,-94\\" style=\\"stroke-dasharray: 295.3097094374406px 295.3097094374406px; stroke-dashoffset: -0px; transition: stroke-dashoffset .3s ease 0s, stroke-dasharray .3s ease 0s, stroke .3s;\\"></path>
    <path class=\\"ix-progress-circle-path ix-progress-circle-bg\\" fill-opacity=\\"0\\" stroke-linecap=\\"round\\" stroke-width=\\"6\\" d=\\"M 50,50 m 0,-47
       a 47,47 0 1 1 0,94
       a 47,47 0 1 1 0,-94\\" style=\\"transition: stroke-dashoffset .3s ease 0s, stroke-dasharray .3s ease 0s, stroke .3s, stroke-width .06s ease .3s; stroke-dasharray: 147.6548547187203px 295.3097094374406px; stroke-dashoffset: -0px;\\"></path>
  </svg>
  <div class=\\"ix-progress-text\\">50%</div>
</div>"
`;

exports[`Progress.vue render dashboard progress work 1`] = `
"<div class=\\"ix-progress-circle ix-progress-status-normal ix-progress\\" style=\\"width: 132px; height: 132px; font-size: 25.8px;\\"><svg viewBox=\\"0 0 100 100\\">
    <!--v-if-->
    <path class=\\"ix-progress-circle-trail\\" stroke=\\"#f5f5f5\\" fill-opacity=\\"0\\" stroke-linecap=\\"round\\" stroke-width=\\"6\\" d=\\"M 50,50 m 0,47
       a 47,47 0 1 1 0,-94
       a 47,47 0 1 1 0,94\\" style=\\"stroke-dasharray: 220.30970943744057px 295.3097094374406px; stroke-dashoffset: -37.5px; transition: stroke-dashoffset .3s ease 0s, stroke-dasharray .3s ease 0s, stroke .3s;\\"></path>
    <path class=\\"ix-progress-circle-path ix-progress-circle-bg\\" fill-opacity=\\"0\\" stroke-linecap=\\"round\\" stroke-width=\\"6\\" d=\\"M 50,50 m 0,47
       a 47,47 0 1 1 0,-94
       a 47,47 0 1 1 0,94\\" style=\\"transition: stroke-dashoffset .3s ease 0s, stroke-dasharray .3s ease 0s, stroke .3s, stroke-width .06s ease .3s; stroke-dasharray: 110.15485471872029px 295.3097094374406px; stroke-dashoffset: -37.5px;\\"></path>
  </svg>
  <div class=\\"ix-progress-text\\">50%</div>
</div>"
`;

exports[`Progress.vue render line progress work 1`] = `
"<div class=\\"ix-progress-line ix-progress-status-normal ix-progress-show-info ix-progress-round ix-progress\\">
  <div class=\\"ix-progress-outer\\">
    <div class=\\"ix-progress-inner\\">
      <!--v-if-->
      <div class=\\"ix-progress-bg\\" style=\\"height: 8px; width: 50%;\\"></div>
    </div>
  </div>
  <div class=\\"ix-progress-text\\">50%</div>
</div>"
`;

exports[`Progress.vue render successColor progress 1`] = `
"<div class=\\"ix-progress-line ix-progress-status-normal ix-progress-show-info ix-progress-round ix-progress\\">
  <div class=\\"ix-progress-outer\\">
    <div class=\\"ix-progress-inner\\">
      <div class=\\"ix-progress-success-bg\\" style=\\"height: 8px; width: 20%; background: red;\\"></div>
      <div class=\\"ix-progress-bg\\" style=\\"height: 8px; width: 50%;\\"></div>
    </div>
  </div>
  <div class=\\"ix-progress-text\\">50%</div>
</div>"
`;

exports[`Progress.vue strokeColor work 1`] = `
"<div class=\\"ix-progress-line ix-progress-status-normal ix-progress-show-info ix-progress-round ix-progress\\">
  <div class=\\"ix-progress-outer\\">
    <div class=\\"ix-progress-inner\\">
      <!--v-if-->
      <div class=\\"ix-progress-bg\\" style=\\"height: 8px; width: 50%; background: red;\\"></div>
    </div>
  </div>
  <div class=\\"ix-progress-text\\">50%</div>
</div>"
`;

exports[`Progress.vue strokeColor work 2`] = `
"<div class=\\"ix-progress-circle ix-progress-status-normal ix-progress-circle-gradient ix-progress\\" style=\\"width: 132px; height: 132px; font-size: 25.8px;\\"><svg viewBox=\\"0 0 100 100\\">
    <defs>
      <linearGradient id=\\"ix-progress-gradient-ix-2\\" x1=\\"100%\\" y1=\\"0%\\" x2=\\"0%\\" y2=\\"0%\\">
        <stop offset=\\"0%\\" stop-color=\\"#108ee9\\"></stop>
        <stop offset=\\"100%\\" stop-color=\\"#87d068\\"></stop>
      </linearGradient>
    </defs>
    <path class=\\"ix-progress-circle-trail\\" stroke=\\"#f5f5f5\\" fill-opacity=\\"0\\" stroke-linecap=\\"round\\" stroke-width=\\"6\\" d=\\"M 50,50 m 0,-47
       a 47,47 0 1 1 0,94
       a 47,47 0 1 1 0,-94\\" style=\\"stroke-dasharray: 295.3097094374406px 295.3097094374406px; stroke-dashoffset: -0px; transition: stroke-dashoffset .3s ease 0s, stroke-dasharray .3s ease 0s, stroke .3s;\\"></path>
    <path stroke=\\"url(#ix-progress-gradient-ix-2)\\" class=\\"ix-progress-circle-path\\" fill-opacity=\\"0\\" stroke-linecap=\\"round\\" stroke-width=\\"6\\" d=\\"M 50,50 m 0,-47
       a 47,47 0 1 1 0,94
       a 47,47 0 1 1 0,-94\\" style=\\"transition: stroke-dashoffset .3s ease 0s, stroke-dasharray .3s ease 0s, stroke .3s, stroke-width .06s ease .3s; stroke-dasharray: 147.6548547187203px 295.3097094374406px; stroke-dashoffset: -0px;\\"></path>
  </svg>
  <div class=\\"ix-progress-text\\">50%</div>
</div>"
`;

exports[`Progress.vue strokeColor work 3`] = `
"<div class=\\"ix-progress-line ix-progress-status-normal ix-progress-show-info ix-progress-round ix-progress\\">
  <div class=\\"ix-progress-outer\\">
    <div class=\\"ix-progress-inner\\">
      <!--v-if-->
      <div class=\\"ix-progress-bg\\" style=\\"height: 8px; width: 50%; background: red;\\"></div>
    </div>
  </div>
  <div class=\\"ix-progress-text\\">50%</div>
</div>"
`;

exports[`Progress.vue strokeColor work 4`] = `
"<div class=\\"ix-progress-line ix-progress-status-normal ix-progress-show-info ix-progress-round ix-progress\\">
  <div class=\\"ix-progress-outer\\">
    <div class=\\"ix-progress-inner\\">
      <!--v-if-->
      <div class=\\"ix-progress-bg\\" style=\\"height: 8px; width: 50%; background: red;\\"></div>
    </div>
  </div>
  <div class=\\"ix-progress-text\\">50%</div>
</div>"
`;

exports[`Progress.vue trailColor work 1`] = `
"<div class=\\"ix-progress-line ix-progress-status-normal ix-progress-show-info ix-progress-round ix-progress\\">
  <div class=\\"ix-progress-outer\\">
    <div class=\\"ix-progress-inner\\">
      <!--v-if-->
      <div class=\\"ix-progress-bg\\" style=\\"height: 8px; width: 50%;\\"></div>
    </div>
  </div>
  <div class=\\"ix-progress-text\\">50%</div>
</div>"
`;
